### PR TITLE
Use dmVMath instead of Vectormath::Aos

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/ddf/src/ddf/ddf_math.proto [hunk 1]
```diff
@@ -5,11 +5,11 @@ import "ddf/ddf_extensions.proto";
 option java_package = "com.dynamo.proto";
 option java_outer_classname = "DdfMath";
 
-option (ddf_includes) = "dmsdk/vectormath/cpp/vectormath_aos.h";
+option (ddf_includes) = "dmsdk/dlib/vmath.h";
```
Comment (JCash) [inline engine/ddf/src/ddf/ddf_math.proto:8]: The goal is to at least open up for the possibility of using another vector library.
The current one doesn't support iOS/Android with NEON simd.

Also, having a shorter namespace is good imho.

File: engine/dlib/src/dlib/transform.h [hunk 2]
```diff
@@ -29,13 +27,13 @@ namespace dmTransform
     class TransformS1
     {
         // tx, ty, tz, scale
-        Vector4 m_TranslationScale;
-        Quat    m_Rotation;
+        dmVMath::Vector4 m_TranslationScale;
+        dmVMath::Quat    m_Rotation;
```
Comment (JCash) [inline engine/dlib/src/dlib/transform.h:31]: We never want to "open up" namespaces in a header file, even in another namespace.
If makes the "ownership" unclear, and is may hinder refactoring etc.

I.e. `dmTransform::Vector4` shouldn't be allowed. Neither should "Vector4" (if it was added to the global scope)

File: engine/dlib/src/test/test_transform.cpp [hunk 3]
```diff
@@ -15,7 +15,7 @@
 #include "dlib/transform.h"
 #include "dlib/math.h"
 
-using namespace Vectormath::Aos;
+using namespace dmVMath;
```
Comment (JCash) [inline engine/dlib/src/test/test_transform.cpp:18]: In tests (a single translation unit), it's more convenient to add it at global namespace scope.

File: engine/engine/src/engine.cpp [hunk 4]
```diff
@@ -106,6 +104,9 @@ extern "C" {
 
 namespace dmEngine
 {
+
+    using namespace dmVMath;
```
Comment (JCash) [inline engine/engine/src/engine.cpp:108]: In cpp files, we may open up the namespace if there are a lot of uses of the types/functions.

File: engine/gui/src/test/test_gui_script.cpp [hunk 5]
```diff
@@ -841,20 +844,20 @@ TEST_F(dmGuiScriptTest, TestCancelAnimation)
...
 
-        Vectormath::Aos::Matrix4 t2;
+        dmVMath::Matrix4 t2;
         const float tinyDifference = 10e-10f;
         while (ticks < 8) {
             dmGui::RenderScene(scene, RenderNodesStoreTransform, &t1);
             dmGui::UpdateScene(scene, 0.125f);
-            Vectormath::Aos::Vector3 currentDiagonal = Vector3(t1[0][0], t1[1][1], t1[2][2]);
+            dmVMath::Vector3 currentDiagonal = Vector3(t1[0][0], t1[1][1], t1[2][2]);
             if (tinyDifference < Vectormath::Aos::lengthSqr(currentDiagonal - postScaleDiagonal)) {
```
Comment (JCash) [inline engine/gui/src/test/test_gui_script.cpp:861]: Currently, only the types are added to the `dmVMath` namespace.
Another task will be to add the commonly used functions as well.

**Review by britzl — APPROVED** [review]

**Pull request merged**